### PR TITLE
Break apart ros_subscriber test translation unit

### PR DESCRIPTION
**Pull request by mjcarroll**
Another attempt at reducing compilation memory usage.

Previously, building `ros_subscriber.cpp` used about 1.1 GB of RAM according to `templight`.

After splitting the individual TUs use:

image

### CODE REVIEW

File: ros_ign_bridge/test/subscribers/ros_subscriber/geometry_msgs_subscriber.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+// limitations under the License.
+//
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
```
Comment (chapulina) [inline ros_ign_bridge/test/subscribers/ros_subscriber/geometry_msgs_subscriber.cpp]: Do we need `rclcpp` on these cpp files?

File: ros_ign_bridge/test/subscribers/ros_subscriber/nav_msgs_subscriber.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// limitations under the License.
+//
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
```
Comment (chapulina) [inline ros_ign_bridge/test/subscribers/ros_subscriber/nav_msgs_subscriber.cpp]: I think we can remove `rclcpp` and `string`

**Review by chapulina — APPROVED** [review]
LGTM, I think we can get rid of some includes too for extra optimization

File: ros_ign_bridge/test/subscribers/ros_subscriber/geometry_msgs_subscriber.cpp (hunk 1 above)
Comment (mjcarroll) [reply ros_ign_bridge/test/subscribers/ros_subscriber/geometry_msgs_subscriber.cpp]: 49be345

File: ros_ign_bridge/test/subscribers/ros_subscriber/nav_msgs_subscriber.cpp (hunk 2 above)
Comment (mjcarroll) [reply ros_ign_bridge/test/subscribers/ros_subscriber/nav_msgs_subscriber.cpp]: 49be345

**Pull request merged**